android/build.gradle

```diff
@@ -1,69 +1,116 @@
-apply plugin: 'com.android.library'
+buildscript {
+  // Buildscript is evaluated before everything else so we can't use getExtOrDefault
+  def kotlin_version = rootProject.ext.has("kotlinVersion") ? rootProject.ext.get("kotlinVersion") : project.properties["MapLibreReactNative_kotlinVersion"]
+
+  repositories {
+    google()
+    mavenCentral()
+  }
+
+  dependencies {
+    classpath "com.android.tools.build:gradle:7.2.1"
+    // noinspection DifferentKotlinGradleVersion
+    classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+  }
+}
 
-def safeExtGet(prop, fallback) {
-    rootProject.ext.has(prop) ? rootProject.ext.get(prop) : fallback
+def reactNativeArchitectures() {
+  def value = rootProject.getProperties().get("reactNativeArchitectures")
+  return value ? value.split(",") : ["armeabi-v7a", "x86", "x86_64", "arm64-v8a"]
 }
 
 def isNewArchitectureEnabled() {
-    return project.hasProperty("newArchEnabled") && project.newArchEnabled == "true"
+  return rootProject.hasProperty("newArchEnabled") && rootProject.getProperty("newArchEnabled") == "true"
 }
 
-buildscript {
-    ext {
-        buildToolsVersion = "33.0.0"
-        minSdkVersion = 16
-        compileSdkVersion = 34
-        targetSdkVersion = 33
-        supportLibVersion = "28.0.0"
-    }
+apply plugin: "com.android.library"
+apply plugin: "kotlin-android"
 
-    repositories {
-        google()
-        mavenCentral()
-    }
+// TODO: Currently breaks new arch support, but works without
```

Comment:
Are we intending to merge this in if it doesn't work with new arch?

Reply:
It works for new arch, if the plugin is not applied! So I want to merge it in the current state, but keep the comment in, as it would actually be used in the scaffold. We want to un-comment this at some point, when we migrate further towards new architecture.